### PR TITLE
 adoptopenjdk-bin: init at 11

### PR DESCRIPTION
**Pull request by taku0**
###### Motivation for this change

Oracle JDK 11 is not free of charge (except for development, testing, prototyping, or demonstration), and free users are guided to use OpenJDK 11. `openjdk11` package is in progress, but not building yet (https://github.com/NixOS/nixpkgs/pull/47375#issuecomment-428671180).

This PR add binary packages of AdoptOpenJDK, which includes combinations of JDK/JRE and Hotspot/OpenJ9 variants.

Only `x86_64-linux` is supported for now, though AdoptOpenJDK itself supports many platforms and architectures.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: lib/licenses.nix [hunk 1]
```diff
@@ -309,6 +309,12 @@ lib.mapAttrs (n: v: v // { shortName = n; }) rec {
     fullName = "GNU General Public License v2.0 only";
   };
 
+  gpl2Classpath = {
+    spdxId = "GPL-2.0 WITH Classpath-exception-2.0";
```
Comment (matthewbauer) [inline lib/licenses.nix]: Identifier should be GPL-2.0-with-classpath-exception

Comment (taku0) [reply lib/licenses.nix]: Fixed.

**Comment by taku0** [comment]
Added Darwin packages.

**Review by c0bw3b — APPROVED** [review]
Green on nixos-18.09

```shell
Result in /[...]/nox-review-7i4pr45u
total 0
lrwxrwxrwx 1 zzzz users 71 24 oct.  18:46 result -> /nix/store/dl1psv9krb2ma1pl5y7s8ll41w3fwbwq-adoptopenjdk-hotspot-bin-11
lrwxrwxrwx 1 zzzz users 75 24 oct.  18:46 result-2 -> /nix/store/md4fhdpizm40lqps9501p06kxj4714y3-adoptopenjdk-jre-hotspot-bin-11
lrwxrwxrwx 1 zzzz users 74 24 oct.  18:46 result-3 -> /nix/store/vijvm03n69sfb0g4p3ms7smpg45lp359-adoptopenjdk-jre-openj9-bin-11
lrwxrwxrwx 1 zzzz users 70 24 oct.  18:46 result-4 -> /nix/store/dqvq26h92j1yc1vzmzprdpwmqld27iqj-adoptopenjdk-openj9-bin-11
```

**Comment by c0bw3b** [comment]
This is good to go right?
cc @NeQuissimus @fpletz as maintainers of other JDKs

**Comment by NeQuissimus** [comment]
LGTM

**Review by NeQuissimus — APPROVED** [review]

**Pull request merged**